### PR TITLE
Add getopt package

### PR DESCRIPTION
**Pull request by caseymcc**
hunterize wingetopt and change name to getopt

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,44 @@
-PROJECT(wingetopt)
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.0)
+
+project(getopt VERSION 1.0.0)
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src)
-add_library(wingetopt STATIC src/getopt.c src/getopt.h)
+add_library(getopt STATIC src/getopt.c src/getopt.h)
```
Comment (ruslo) [inline CMakeLists.txt:6]: Why `STATIC`? Is it designed to not be used as shared? See:
* http://cgold.readthedocs.io/en/latest/tutorials/libraries/static-shared.html

File: Config.cmake.in [hunk 2]
```diff
@@ -0,0 +1 @@
+include("${CMAKE_CURRENT_LIST_DIR}/@targets_export_name@.cmake")
```
Comment (ruslo) [inline Config.cmake.in:3]: You should add `check_required_components`. In this case if user wrongly call `find_package(getopt REQUIRED nonexistent_component)` the error will be reported. See https://github.com/forexample/package-example/blob/cf2ea1d6a209fb9eca2ab83fdd0ac15fe4d3e807/Foo/cmake/Config.cmake.in

**Review by ruslo — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (caseymcc) [reply CMakeLists.txt:6]: It is not setup for dynamic builds, would need some _dllexports for that. Could be added but considering it is only 2 functions likely not worth it.

File: Config.cmake.in (hunk 2 above)
Comment (caseymcc) [reply Config.cmake.in:3]: Ill defer to you on the cmake config stuff, haven't really worked with it before using hunter.

**Comment by caseymcc** [comment]
Added check_required_components to config file. Unless you see something else should be good to go. Haven't done testing on my side yet so if anything pops up I will look into it.

File: Config.cmake.in (hunk 2 above)
Comment (ruslo) [reply Config.cmake.in:3]: As far as I understand `check_required_components` will not work without `@PACKAGE_INIT@`. Just copy/paste the `Config.cmake.in` from the link I provided.

**Review by ruslo — CHANGES_REQUESTED** [review]

**Comment by ruslo** [comment]
https://github.com/ruslo/hunter/blob/v0.23.214/README.rst

**Pull request closed without merging**